SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityUsageHelper.java

```diff
@@ -45,4 +103,446 @@ public SingularityUsageHelper(SlaveManager slaveManager) {
     return slavesToTrack;
   }
 
+  public Optional<SingularitySlaveUsage> collectSlaveUsageSimple(
+      SingularitySlave slave,
+      long now,
+      boolean useShortTimeout) {
+    Optional<Long> memoryMbTotal = Optional.absent();
+    Optional<Double> cpusTotal = Optional.absent();
+    Optional<Long> diskMbTotal = Optional.absent();
+
+    try {
+      List<MesosTaskMonitorObject> allTaskUsage = mesosClient.getSlaveResourceUsage(slave.getHost(), useShortTimeout);
+      MesosSlaveMetricsSnapshotObject slaveMetricsSnapshot = mesosClient.getSlaveMetricsSnapshot(slave.getHost());
+
+      double memoryMbReservedOnSlave = 0L;
+      double cpuReservedOnSlave = 0.0;
+      double diskMbReservedOnSlave = 0L;
+
+      double systemMemTotalBytes = 0;
+      double systemMemFreeBytes = 0;
+      double systemLoad1Min = 0;
+      double systemLoad5Min = 0;
+      double systemLoad15Min = 0;
+      double slaveDiskUsed = 0;
+      double slaveDiskTotal = 0;
+      double systemCpusTotal = 0;
+      if (slaveMetricsSnapshot != null) {
+        memoryMbReservedOnSlave = slaveMetricsSnapshot.getSlaveMemUsed();
+        cpuReservedOnSlave = slaveMetricsSnapshot.getSlaveCpusUsed();
+        diskMbReservedOnSlave = slaveMetricsSnapshot.getSlaveDiskUsed();
+        systemMemTotalBytes = slaveMetricsSnapshot.getSystemMemTotalBytes();
+        systemMemFreeBytes = slaveMetricsSnapshot.getSystemMemFreeBytes();
+        systemLoad1Min = slaveMetricsSnapshot.getSystemLoad1Min();
+        systemLoad5Min = slaveMetricsSnapshot.getSystemLoad5Min();
+        systemLoad15Min = slaveMetricsSnapshot.getSystemLoad15Min();
+        slaveDiskUsed = slaveMetricsSnapshot.getSlaveDiskUsed();
+        slaveDiskTotal = slaveMetricsSnapshot.getSlaveDiskTotal();
+        systemCpusTotal = slaveMetricsSnapshot.getSystemCpusTotal();
+      }
+
+      if (!slave.getResources().isPresent() ||
+          !slave.getResources().get().getMemoryMegaBytes().isPresent() ||
+          !slave.getResources().get().getNumCpus().isPresent()) {
+        LOG.debug("Could not find slave or resources for slave {}", slave.getId());
+      } else {
+        memoryMbTotal = Optional.of(slave.getResources().get().getMemoryMegaBytes().get().longValue());
+        cpusTotal = Optional.of(slave.getResources().get().getNumCpus().get().doubleValue());
+        diskMbTotal = Optional.of(slave.getResources().get().getDiskSpace().get());
+      }
+
+      SingularitySlaveUsage slaveUsage = new SingularitySlaveUsage(
+          cpuReservedOnSlave, cpuReservedOnSlave, cpusTotal,
+          memoryMbReservedOnSlave, memoryMbReservedOnSlave, memoryMbTotal,
+          diskMbReservedOnSlave, diskMbReservedOnSlave, diskMbTotal,
```

Comment:
Seems pretty funky to send in these same variables, but that's what reducing everything to use the metric snapshot resulted in (see the plain `collectSlaveUsage` below to compare).

Reply:
I guess the 'clean' way to do it could be to have a separate POJO with a more minimal set of fields, which gets used in the scheduling. The fuller class, which would extend that, has all of the fields is used in the poller + api

e.g. SingularitySlaveUsage extends SingularitySimpleSlaveUsage {}